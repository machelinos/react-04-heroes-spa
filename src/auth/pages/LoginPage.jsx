import { useContext } from "react";
import { useNavigate } from "react-router-dom";
import { AuthContext } from "../context";

export const LoginPage = () => {

  const navigate = useNavigate();

  const {login} = useContext(AuthContext);

  const onLogin = () => {

    login({
      id: 'ABC',
      name: 'Marcel Cabrera'
    });
    
    navigate('/',{
      replace: true
    })
  }
  
  return (
    <div className="container mt-2">
        <h1>Login Page</h1>
        <hr />

        <button
          className="btn btn-primary"
          onClick={onLogin}
        >Login</button>

    </div>
  )
}
