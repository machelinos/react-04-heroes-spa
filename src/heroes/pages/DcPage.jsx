import { HeroList } from "../components";
import { getHeroesByPublisher } from "../helpers"

export const DcPage = () => {

  return (
    <>
        <h1>DC Page</h1>
        <hr />

        <HeroList publisher={'DC Comics'} />
    </>
  )
}
