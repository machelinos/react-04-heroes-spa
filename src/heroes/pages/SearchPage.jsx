import { useLocation, useNavigate } from "react-router-dom";
import queryString from "query-string";
import { useForm } from "../../hooks/useForm";
import { useMemo } from "react";
import { getHeroesByName } from "../helpers";
import { HeroCard } from "../components";

export const SearchPage = () => {
  const navigate = useNavigate();
  const location = useLocation();

  const {q = ''} = queryString.parse(location.search);

  const { searchText, onInputChange } = useForm({
    searchText: q
  });

  const heroes = useMemo(()=>{
    return getHeroesByName(searchText);
  },[location]);

  const onSubmitSearch = (event)=>{
    event.preventDefault();
    
    navigate(`?q=${searchText}`);
  }

  return (
    <>
        <h1>Search</h1>
        <hr />

        <div className="row">
          <div className="col-5">
            <form onSubmit={onSubmitSearch}>
              <input
                autoComplete="off"
                className="form-control"
                name="searchText"
                placeholder="Search hero..."
                type="text"
                value={searchText}
                onChange={onInputChange}
              />

              <button
                className="btn btn-outline-primary mt-2"
              >
                Search
              </button>
            </form>

          </div>

          <div className="col-7">
            <h4>Results</h4>
            <hr />

            {
              !q && (
                <div className="alert alert-primary">
                  Search a hero
                </div>
  
              )
            }

            {
              (q && heroes.length===0) && (
                <div className="alert alert-danger">
                  There is no hero <strong>{ q }</strong>
                </div>
              )
            }

            {
              (q && heroes.length > 0) && (
                  heroes.map(hero=>(
                      <HeroCard
                          key={hero.id}
                          {...hero}
                      />
                  ))
              )
            }

          </div>

        </div>

    </>
  )
}
