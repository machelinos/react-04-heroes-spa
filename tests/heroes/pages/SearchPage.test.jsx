import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SearchPage } from "../../../src/heroes/pages/SearchPage";

describe('Search page tests', ()=>{
    test('must load initial default data', ()=>{
        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });
});