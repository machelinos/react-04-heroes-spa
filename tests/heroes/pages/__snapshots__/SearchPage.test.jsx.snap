// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search page tests must load initial default data 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search hero..."
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
